secio/src/dh_compat/openssl_impl.rs

```diff
@@ -0,0 +1,177 @@
+use openssl::{
+    bn::BigNumContext,
+    derive::Deriver,
+    ec::{EcGroup, EcKey, EcPoint, PointConversionForm},
+    nid::Nid,
+    pkey::{Id, PKey, Private, Public},
+};
+
+use crate::{dh_compat::KeyAgreement, error::SecioError};
+
+struct Algorithm {
+    _private_len: usize,
+    pubkey_len: usize,
+    pairs_generate: fn() -> Result<(PKey<Private>, Vec<u8>), SecioError>,
```

Comment:
```suggestion
    pairs_generate: fn() -> Result<(PKey<Private>, PKey<Public>), SecioError>,
```

Reply:
We don't need the pubkey structure to be generated at the same time, all we need is its corresponding bytes to send to the remote end, so we don't need to expose the internal structure here